students/mark/Session05/mailroom.py

```diff
@@ -0,0 +1,267 @@
+The script should accomplish the following goals:
+
+    It should have a data structure that holds a list of your donors and a history of the
+    amounts they have donated. This structure should be populated at first with at least
+    five donors, with between 1 and 3 donations each.
+
+    You can store that data structure in the global namespace.
+
+    The script should prompt the user (you) to choose from a menu of 3 actions:
+    “Send a Thank You”, “Create a Report” or “quit”)
+
+
+
+### Finally, use only functions and the basic Python data types you’ve learned about so far.
+There is no need to go any farther than that for this assignment.
+
+/*
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------
+William Gates, III         $  653784.49           2  $   326892.24
+Mark Zuckerberg            $   16396.10           3  $     5465.37
+Jeff Bezos                 $     877.33           1  $      877.33
+Paul Allen                 $     708.42           3  $      236.14
+*/
+
+/*
+
+
+*/
+"""
+
+# donor_db = {"William Gates, III":[326888.82, 326895.67],
+#             "Mark Zuckerberg":[5565.37, 5465.37, 5365.36],
+#             "Jeff Bezos":[877.33],
+#             "paul allen":[236.14, 236.14, 236.14],
+#             "jose gonzalez" : [123.45, 678.90, 101.11]
+#             }
+
+donor_db = {"william gates, iii":[326888.82, 326895.67],
+            "mark zuckerberg":[5565.37, 5465.37, 5365.36],
+            "jeff bezos":[877.33],
+            "paul allen":[236.14, 236.14, 236.14],
+            "jose gonzalez" : [123.45, 678.90, 101.11]
+            }
+
+
+def format_donor_name(donor_names):
+    if ',' in donor_names:
+        name_title=donor_names.split(',')
+        i_var=name_title[0].title() + ','
+        donor_names=i_var.replace(',', "," + name_title[1].upper())
+        return(donor_names)
+    else:
+        name_title=donor_names
+        return(donor_names.title())
+
+
+def print_donor_name(donor_db):
+    """
+    Output the donor name(s)
+    :param: donor_db dict
+    :returns: zero for success
+    """
+    print("Donor list: \n")
+    for donor_names in donor_db.keys():
+        if ',' in donor_names:
+            name_title=donor_names.split(',')
+            i_var=name_title[0].title() + ','
+            donor_names=i_var.replace(',', "," + name_title[1].upper())
+            print(donor_names)
+        else:
+            name_title=donor_names
+            print(donor_names.title())
+    return 0
+
+
+def get_donor(name):
+    """
+    get a donor out of the donor "structure"
+    :params: name/
+    :return: donor
+    """
+    for donor in donor_db:
+        if name.strip().lower() == donor[0].lower():
+            """ get the name in lowercase, makes matching names easier later"""
+            return donor
+
+    return None # or should this be donor?
```

Comment:
You don't want to return donor in this case, because that would return whatever the last donor in donor_db is, regardless of whether it is a match. You don't have to return anything here. If you get all the way through donor_db and don't find a match and return it, then python will return None by default. Again, the end of the function is defined by the white space. Weird, I know, but you get used to it. :)